types/hexo/index.d.ts

```diff
@@ -667,7 +667,12 @@ declare namespace Hexo {
         }
 
         interface Generator {
-            register(name: string, fn: (locals: Site) => Generator.Return | Generator.Return[]): void;
+            register(
+                name: string,
+                fn: (
+                    locals: Site,
+                ) => Generator.Return | Generator.Return[] | Promise<Generator.Return> | Promise<Generator.Return[]>,
```

Comment:
I'm not sure this does work. Under the hood, Hexo calls bluebird's `Promise.method` on the function which I think results in the error:

```
Uncaught TypeError: expecting a function but got [object Promise]
```

https://github.com/hexojs/hexo/blob/92b979f4a3fa8714aebd3d11c3295d466b870905/lib/extend/generator.js#L29

Reply:
Fixed. thank you!